src/utils/canvas.js

```diff
@@ -155,24 +155,164 @@ export const generateHalfMountainPoints = (
+  moleculeCenter,
+  oxygenRadius,
+  hydrogenRadius,
+) => {
+  const { x: moleculeCenterX, y: moleculeCenterY } = moleculeCenter;
+  const hydrogenXOffset = Math.sin((45 * Math.PI) / 180) * oxygenRadius;
+  const hydrogenYOffset =
+    Math.cos((45 * Math.PI) / 180) * oxygenRadius + hydrogenRadius;
+  return {
+    topHydrogen: {
+      x: moleculeCenterX - hydrogenXOffset,
+      y: moleculeCenterY - hydrogenYOffset,
+    },
+    oxygen: { x: moleculeCenterX, y: moleculeCenterY },
+    bottomHydrogen: {
+      x: moleculeCenterX - hydrogenXOffset,
+      y: moleculeCenterY + hydrogenYOffset,
+    },
+  };
+};
 
-  const rowOne = [
-    { x: treeBaseMidpoint + halfRadius, y: rowOneY },
-    { x: treeBaseMidpoint - halfRadius, y: rowOneY },
-  ];
-  const rowTwo = [{ x: treeBaseMidpoint, y: rowTwoY }];
+export const determineMethaneAtomsCoordinates = (
+  moleculeCenter,
+  carbonRadius,
+  hydrogenRadius,
+) => {
+  const { x: moleculeCenterX, y: moleculeCenterY } = moleculeCenter;
+
+  const leftHydrogensXOffset = Math.cos((30 * Math.PI) / 180) * carbonRadius;
+  const leftHydrogensYOffset =
+    Math.sin((30 * Math.PI) / 180) * carbonRadius + hydrogenRadius;
+  const rightHydrogensXOffset = Math.cos((40 * Math.PI) / 180) * carbonRadius;
+  const rightHydrogensYOffset =
+    Math.sin((40 * Math.PI) / 180) * carbonRadius + hydrogenRadius;
+
+  return {
+    carbon: { x: moleculeCenterX, y: moleculeCenterY },
+    topLeftHydrogen: {
+      x: moleculeCenterX - leftHydrogensXOffset,
+      y: moleculeCenterY - leftHydrogensYOffset,
+    },
+    topRightHydrogen: {
+      x: moleculeCenterX + rightHydrogensXOffset,
+      y: moleculeCenterY - rightHydrogensYOffset,
+    },
+    bottomRightHydrogen: {
+      x: moleculeCenterX + rightHydrogensXOffset,
+      y: moleculeCenterY + rightHydrogensYOffset,
+    },
+    bottomLeftHydrogen: {
+      x: moleculeCenterX - leftHydrogensXOffset,
+      y: moleculeCenterY + leftHydrogensYOffset,
+    },
+  };
+};
+
+const determineNumberOfMolecules = (moleculeDistribution) => {
+  return Object.values(moleculeDistribution).reduce(
+    (accumulator, currentValue) => accumulator + currentValue,
+    0,
+  );
+};
+
+export const determineMoleculeCenterXs = (
+  moleculeRowBeginsX,
+  centralAtomRadius,
+  spaceBetweenMolecules,
+  moleculeDistribution,
+) => {
+  const numberOfMolecules = determineNumberOfMolecules(moleculeDistribution);
+  const centerXs = new Array(numberOfMolecules)
+    .fill()
+    .map(
+      (emptyElement, index) =>
+        moleculeRowBeginsX +
+        (2 * index + 1) * centralAtomRadius +
+        index * spaceBetweenMolecules,
+    );
+  return centerXs;
+};
+
+export const distributeMoleculesRandomly = (moleculeDistribution) => {
+  const numberOfMolecules = determineNumberOfMolecules(moleculeDistribution);
+  const randomDistribution = new Array(numberOfMolecules);
+  const moleculeDistributionArray = Object.entries(moleculeDistribution);
```

Comment:
`_.shuffle()` might be the solution you were looking for? This is interesting too https://flaviocopes.com/how-to-shuffle-array-javascript/